core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1154,6 +1176,7 @@ private[spark] class DAGScheduler(
     // Use the scheduling pool, job group, description, etc. from an ActiveJob associated
     // with this Stage
     val properties = jobIdToActiveJob(jobId).properties
+    addPysparkConfigsToProperties(stage, properties)
```

Comment:
I'm not aware of how `BasePythonRunner` works, but will pyspark properties propagated from different stages conflict with each other? e.g. stage1 and stage2 are running concurrently, then, task2.0 may happen to reads task1.0's properties. Is it possible?

Reply:
no. The properties here are a local to this call and are added here when the tasks for this stage are created, see down further line 1279ish. the localproperties get propogated to the TaskContext and the pythonrunner looks at each task for these values.